### DISCUSSION TITLE
pg_connect crash when connecting to PostgreSQL

### DISCUSSION THREAD

**Discussion by ataklychev** [shivammathur/homebrew-php#2244] [Q&A / Help]
**Describe the bug**
PHP-FPM crash when connecting to PostgreSQL

**PHP versions**
8.3.1

**To Reproduce**
On MacOS sonoma 14.2.1
brew install postgresql@16 ( create test db )
brew services start postgresql@16
brew install shivammathur/php/php@8.3
brew services start php
brew install nginx ( configure simple server with fastcgi_pass 127.0.0.1:9000 )
brew services start nginx

In console mode, pg_connect work perfectly. In fpm mode pg_connect crashes and nginx return 502.

I found similar issue https://github.com/Homebrew/homebrew-core/issues/155651 , and try to downgraded openssl to 3.1.4.

But when i install php [brew install shivammathur/php/php@8.3], openssl 3.2 installed force again and php use openssl 3.2.

How can i downgrade openssl to 3.1.4 in php ?

**Comment by shivammathur** [comment]
@ataklychev

This was patched on Homebrew's side to work with openssl 3.2.0, Please reinstall your `postgresql@16` and `libpq` formulae.
```zsh
brew update
brew reinstall libpq postgresql@16
```

**Comment by ataklychev** [comment]
> @ataklychev
> 
> This was patched on Homebrew's side to work with openssl 3.2.0, Please reinstall your `postgresql@16` and `libpq` formulae.
> 
> ```shell
> brew update
> brew reinstall libpq postgresql@16
> ```

It did not help

**Comment by kennethgerrits** [comment]
I'm facing the same issue. Would like to know updates in the future

**Comment by bio** [comment]
@ataklychev @kennethgerrits As a workaround you can disable `gssencmode` in the postgresql data source name (DSN) in your app when connecting to postgresql.
Example: `pgsql:host=<HOST>;port=<PORT>;dbname=<DBNAME>;gssencmode=disable`

**Reply by ataklychev** [reply]
This solves my problem, thank you very much!

**Reply by MacDada** [reply]
`gssencmode=disable` helped me (temporarily?) as well.

Anyone knows a proper fix? Do we need to wait for some bugfix update?

**Reply by MacDada** [reply]
https://github.com/php/php-src/issues/13339
